Refresh terminal section, add some features (#4574)
microsoft/vscode-docs: docs/editor/integrated-terminal.md

![Terminal](images/integrated-terminal/integrated-terminal.png)

> **Note:** Open an external terminal with the `kb(workbench.action.terminal.openNativeConsole)` keyboard shortcut if you prefer to work outside VS Code.

## Managing terminals

> The terminal tabs view is to the right of the terminals. Each terminal has an entry with its name, icon, color, and group decoration (if any).

Reviewer: ```suggestion
The terminal tabs view is on the right side of the terminal view. Each terminal has an entry with its name, icon, color, and group decoration (if any).
```

![Terminal tabs](images/integrated-terminal/tabs.png)
> **Tip:** Change the tabs location using the `integrated.terminal.tabs.location` setting.

Terminal instances can be added by clicking the plus icon on the top-right of the **TERMINAL** panel, selecting a profile from the new terminal dropdown, or by triggering the `kb(workbench.action.terminal.new)` command. This action creates another entry in the tab list associated with that terminal.

Remove terminal instances by hovering a tab and pressing the trash can button, selecting a tab item and pressing `kbstyle(delete)`, by triggering the `kb(workbench.action.terminal.kill)` command, or via the right click context menu.

Navigate between terminal groups using focus next `kbstyle(workbench.action.terminal.focusNext)` and focus previous `kbstyle(workbench.action.terminal.focusPrevious)`.

> Icons may appear to the right of the terminal title on the tab label when the terminal's status change. Statuses include disconnected, in need of relaunch, and bell. Hover the icon to read status information and to take action.

Reviewer: Let's mention tasks status and bell here as examples to demonstrate what statuses are all about

>and to take action

which may contain actions.

### Grouping

Split the terminal by:
- triggering the `kb(workbench.action.terminal.split)` command
- right clicking the context menu and selecting the `Split...` submenu option

> - `kbstyle(alt)` clicking a tab, the plus button, or the the single tab on the terminal panel

Reviewer: Missing clicking the inline/panel split button which will probably be the main way. Let's move the command to the bottom as it's least important.

Navigate between terminals in a group by focusing the previous pane, `kb(workbench.action.terminal.focusPreviousPane)`, and focusing the next pane, `kb(workbench.action.terminal.focusNextPane)`.

Drag and drop tabs to re-arrange the groups.

> Unsplit a split terminal by triggering the `kb(workbench.action.terminal.unsplit)` command.

Reviewer: Suggestion:

Tabs support drag and drop to allow re-arranging, dragging an entry in a terminal group into the empty will remove it from the group (ie. unsplit), dragging a tab into the main terminal area allows joining a group.

Can remove the command part imo

> ### Customization

Reviewer: Customizing Tabs?

    // for tasks and debug
    "terminal.integrated.automationShell.osx": "/bin/sh"
}
```
## Defining the working directory

By default, the terminal will open at the folder that is opened in the Explorer. The `terminal.integrated.cwd` setting allows specifying a custom path to open instead:

```json
{
    "terminal.integrated.cwd": "/home/user"
}
```

Split terminals on Windows will start in the directory that the parent terminal started with. On macOS and Linux, split terminals will inherit the current working directory of the parent terminal. This behavior can be changed using the `terminal.integrated.splitCwd` setting:

```json
{
    "terminal.integrated.splitCwd": "workspaceRoot"
}
```

There are also extensions available that give more options such as [Terminal Here](https://marketplace.visualstudio.com/items?itemName=Tyriar.vscode-terminal-here).

## Terminal process reconnection
Local and remote terminal processes are restored on window reload, such as when an extension install requires a reload. The terminal will be reconnected and the UI state of the terminals will be restored, including the active tab and split terminal relative dimensions.

## Links

> Files and URLs are underlined in the terminal. Low confidence links will be underlined on hover with ctrl/command.

Reviewer: ```suggestion
The terminal features link detection, showing an underline when files or URLs are hovered with the mouse that will go to the target when `kbstyle(Ctrl)`/`kbstyle(Cmd)` is held. If a file or URL cannot be detected, they are still surfaced as "low confidence" links which only show an underline when is held. These low confidence links will search the workspace for the term, opening the match if one is found.
```

Clicking a file link will either open that document in an editor or produce a quick pick with all matches.

> Extensions make use of links in the terminal, such as GitLens, to identify branches.

Reviewer: A screenshot of gitlens hovering over a branch would be good
